src/Generator.cpp

```diff
@@ -1,11 +1,75 @@
+
 #include <cmath>
 #include <fstream>
 #include <set>
+#include <tuple>
+
+#include "llvm/Support/raw_os_ostream.h"  /// for LLVM and STL I/O bridge
+#include "llvm/Support/YAMLTraits.h"      /// for YAML underpinnings
+#include "runtime/HalideRuntime.h"        /// for halide_type_code_t
 
 #include "Generator.h"
 #include "Outputs.h"
 #include "Simplify.h"
 
+namespace Halide {
+namespace Internal {
+
+/// Forward-declare Halide::Internal::YamlEmitter, for use
+/// in both the forward-declaration of its corresponding
+/// llvm::yaml::MappingTraits<T> specialization, and so that
+/// it may be declared as a YAML sequence vector (q.v the
+/// macro LLVM_YAML_IS_SEQUENCE_VECTOR() invocation sub.)
+class YamlEmitter;
+
+} /// namespace Internal
+} /// namespace Halide
+
+namespace llvm {
+namespace yaml {
+
+/// Make the internal GeneratorParam, GeneratorInput,
+/// and GeneratorOutput base types, as well as pointers
+/// to same, available within the llvm::yaml namespace:
+
+using  param_t = Halide::Internal::GeneratorParamBase;
```

Comment:
Aligning the = here I kinda personally like, but it's also contrary to the style used everywhere else in Halide, so please conform to the raggedness we generally use elsewhere.